docs/theories/datatypes.rst

```diff
@@ -109,6 +111,27 @@ Examples
    (assert (and (= (fname x) "John") (= (lname x) "Smith")))
    (check-sat)
 
+Datatype Updaters
+--------------------
+
+Datatype updaters are a (non-standard) extension available in datatype logics.
+The term:
+
+.. code:: smtlib
+   ((_ update Sij) t u)
+
+is equivalent to replacing the field of ``t`` denoted by the selector ``Sij``
+with the value ``u``, or ``t`` itself if that selector does not apply to the
+constructor symbol of ``t``.  For example, for the list datatype, we have that:
```

Comment:
Isn't it easier to say that it doesn't get updated if the selector does not apply?

Reply:
To me, this would suggest that terms are stateful.  Its better to say that the updater returns `t`.